services/src/main/java/org/fao/geonet/api/records/editing/MetadataEditingApi.java

```diff
@@ -378,6 +381,22 @@ public void saveEdits(
 
             // Automatically change the workflow state after save
             if (isEnabledWorkflow) {
+                boolean isAllowedSubmitApproveInvalidMd = sm.getValueAsBool(Settings.METADATA_WORKFLOW_ALLOW_SUBMIT_APPROVE_INVALID_MD);
+                if (((status.equals(StatusValue.Status.SUBMITTED))
+                    || (status.equals(StatusValue.Status.APPROVED)))
+                    && !isAllowedSubmitApproveInvalidMd) {
```

Comment:
I'm not sure about this code, the metadata that is edited should be in `DRAFT` status, no?

In case that makes sense the code, I think can be relevant to check if this other code was executed already: https://github.com/geonetwork/core-geonetwork/pull/7011/files#diff-b08bf8f1e963b6b11841309b167f1acfb7e21895dff6d9c8f9021ebf790b4537L373-R380, to avoid double validation.

Reply:
> I'm not sure about this code, the metadata that is edited should be in DRAFT status, no?

Not always, `status` is an API parameter.

> to avoid double validation.

Indeed. Added a check.